### ISSUE TITLE
Core product CSV importer: .webp cases invalid image error

### ISSUE THREAD

**Issue by jacoswan** [woothemes/woocommerce#37976]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When the image referenced in a CSV import is '.webp', the import fails with the error

```
Invalid image: Sorry, you are not allowed to upload this file type.
```

### Expected behavior

It should support .webp as far as the media library does. 

### Actual behavior

While the test site's media library converted .webp to .jpg (which appears to be the default behavior), installing the Performance Lab Plugin ([mentioned here](https://make.wordpress.org/core/2022/03/28/enabling-webp-by-default/)), enabled the media library to retain the original .webp extension. 

However, the CSV import continues to fail. 

### Steps to reproduce

1. Create a CSV file that references a .webp image in the `Images` column
2. Import that CVS via Products > All Products > Import
3. Notice the error

### WordPress Environment

```
`
### WordPress Environment ###

WordPress address (URL): https://woo.tst
Site address (URL): https://woo.tst
WC Version: 7.6.0
REST API Version: ✔ 7.6.0
WC Blocks Version: ✔ 9.8.4
Action Scheduler Version: ✔ 3.5.4
Log Directory Writable: ✔
WP Version: 6.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.57 (Unix) OpenSSL/1.1.1t PHP/8.2.5
PHP Version: 8.2.5
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 8.0.1
(SecureTransport) OpenSSL/1.1.1t

SUHOSIN Installed: –
MySQL Version: 10.8.3-MariaDB
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.6.0
WC Database Prefix: wp_
Total Database Size: 15.52MB
Database Data Size: 9.71MB
Database Index Size: 5.81MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.17MB + Index: 0.17MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.36MB + Index: 0.38MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.27MB + Index: 0.20MB + Engine InnoDB
wp_afwc_campaigns: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_commission_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_hits: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_payouts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_afwc_payout_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_afwc_referrals: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_customers: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_automatewoo_customer_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_events: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_guests: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_automatewoo_guest_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_log_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_queue: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_automatewoo_referrals: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_automatewoo_referral_advocate_keys: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_automatewoo_referral_invites: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.14MB + Index: 0.09MB + Engine InnoDB
wp_FmeCCFA_fields: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 5.03MB + Index: 0.17MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 1.86MB + Engine InnoDB
wp_posts: Data: 0.30MB + Index: 0.13MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wcpv_commissions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcpv_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wcrp_rental_products_rentals: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_bundle_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_composite_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_warranty_products: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_bis_activity: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_bis_notifications: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_woocommerce_bis_notificationsmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_exported_csv_items: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_gc_activity: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cards: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cardsmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_prl_deploymentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_prl_deployments: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_prl_frequencies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_prl_tracking_conversions: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wpml_mails: Data: 0.05MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

addify_rfq: 1
af_ch_fields: 2
afreg_fields: 1
attachment: 54
aw_workflow: 7
bookable_person: 11
bookable_resource: 2
country_restriction: 1
crf-campaign: 1
crf-contributor: 1
customize_changeset: 3
def_reg_fields: 10
event_ticket: 2
nav_menu_item: 6
order_status: 2
page: 24
post: 2
prl_engine: 1
product: 93
product_variation: 59
revision: 160
shop_coupon: 2
shop_order: 106
shop_order_refund: 2
shop_subscription: 23
sm_fw_suppliers: 1
store-locater: 1
user_request: 2
warranty_request: 2
wbcr-snippets: 17
wc_booking: 29
wc_membership_plan: 1
wcpf_item: 10
wcpf_project: 3
wp_global_styles: 1
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (4) ###

Performance Lab: by WordPress Performance Team – 2.2.0
WooCommerce Beta Tester: by WooCommerce – 2.1.0
WooCommerce Stripe Gateway: by WooCommerce – 7.3.0
WooCommerce: by Automattic – 7.6.0 (update to version 7.7.0-beta.1 is available)

### Inactive Plugins (283) ###

...

### Dropin Plugins (1) ###

object-cache.php: Performance Lab Server Timing Object Cache Drop-In

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
bundle (bundle)
composite (composite)
crf_fund (crf_fund)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)
wcf_funding_product (wcf_funding_product)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
Order datastore: WC_Order_Data_Store_CPT

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.2.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: block-editor-feature-enabled
minified-js
new-product-management-experience
product-variation-management
settings

Daily Cron: ✔ Next scheduled: 2023-04-25 13:31:25 +01:00
Options: ✔
Notes: 68
Onboarding: completed

### Action Scheduler ###

Complete: 634
Oldest: 2023-03-25 12:06:40 +0000
Newest: 2023-04-03 12:15:16 +0100

Failed: 266
Oldest: 2023-02-13 13:35:17 +0000
Newest: 2023-04-18 07:58:38 +0100

Pending: 10
Oldest: 2023-04-26 06:59:25 +0100
Newest: 2024-04-03 12:12:06 +0100


### Status report information ###

Generated at: 2023-04-25 12:32:37 +01:00
```

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by rrennick** [comment]
@jacoswan Thanks for reporting the issue. This should be addressed by #37307. I'll close the issue but please let us know if you still see the issue in the 7.7.0 beta.

**Comment by jacoswan** [comment]
Thanks, @rrennick . Works with 7.7.0-beta.2. Same CSV and image file as for the bug report now imports and converts to .jpg seamlessly and without error 👍🏼
